File: cslacs/Csla/Security/UnauthenticatedIdentity.cs
using System;
using System.Security.Principal;
using Csla.Serialization;
using System.Collections.Generic;
using Csla.Core.FieldManager;
using System.Runtime.Serialization;
using Csla.Core;

namespace Csla.Security
{
  /// <summary>
  /// Implementation of a .NET identity object representing
  /// an unauthenticated user. Used by the
  /// UnauthenticatedPrincipal class.
  /// </summary>
  [Serializable()]
  public sealed class UnauthenticatedIdentity : CslaIdentity
  {
    /// <summary>
    /// Creates an instance of the object.
    /// </summary>
    public UnauthenticatedIdentity()
    {
      IsAuthenticated = false;
      Name = string.Empty;
      AuthenticationType = string.Empty;
      Roles = new MobileList<string>();
    }
  }
}

File: cslavb/Csla/Security/UnauthenticatedIdentity.vb
Imports System
Imports System.Security.Principal
Imports Csla.Serialization
Imports System.Collections.Generic
Imports Csla.Core.FieldManager
Imports System.Runtime.Serialization
Imports Csla.Core

Namespace Security

    ''' <summary>
    ''' Implementation of a .NET identity object representing
    ''' an unauthenticated user. Used by the
    ''' UnauthenticatedPrincipal class.
    ''' </summary>
    <Serializable()> _
    Public NotInheritable Class UnauthenticatedIdentity
        Inherits CslaIdentity

        ''' <summary>
        ''' Creates an instance of the object.
        ''' </summary>
        Public Sub New()
            IsAuthenticated = False
            Name = String.Empty
            AuthenticationType = String.Empty
            Roles = New MobileList(Of String)()
        End Sub

    End Class
End Namespace

